apiserver/facades/agent/caasoperator/watcher_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasoperator_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/facades/agent/caasoperator"
+	"github.com/juju/juju/core/watcher/watchertest"
+	"github.com/juju/juju/state"
+	coretesting "github.com/juju/juju/testing"
+)
+
+var _ = gc.Suite(&IDWatcherSuite{})
+
+type IDWatcherSuite struct {
+	coretesting.BaseSuite
+}
+
+func (s *IDWatcherSuite) TestWatcher(c *gc.C) {
+	m := &mockModel{}
+	m.containers = []state.CloudContainer{
+		&mockCloudContainer{unit: "A", providerID: "a"},
+		&mockCloudContainer{unit: "C", providerID: "c"},
+	}
+	wc := make(chan []string, 4)
+	wc <- []string{"a"}
+	wc <- []string{"b"}
+	wc <- []string{"c"}
+	wc <- []string{"d"}
+	srcWatcher := watchertest.NewMockStringsWatcher(wc)
```

Comment:
wrap this in
testing.NewStringsWatcherC

then you have various assert methods for no change, expected changes etc